### PR TITLE
use QVariantMap instead of QgsStringMap for symbol layer properties

### PR DESCRIPTION
**Pull request by 3nids**

### CODE REVIEW

File: src/core/symbology/qgssymbollayerutils.cpp [hunk 1]
```diff
@@ -2947,33 +2949,49 @@ QgsStringMap QgsSymbolLayerUtils::getVendorOptionList( QDomElement &element )
...
     }
-    e = e.nextSiblingElement();
+    return props;
   }
-  return props;
 }
 
 
-void QgsSymbolLayerUtils::saveProperties( QgsStringMap props, QDomDocument &doc, QDomElement &element )
+void QgsSymbolLayerUtils::saveProperties( QVariantMap props, QDomDocument &doc, QDomElement &element )
```
Comment (3nids) [inline src/core/symbology/qgssymbollayerutils.cpp:2980]: one of the most important changes: we now use QgsXmlUtils to save the properties, which is different from the old behavior. I kept the old saving method to have a backward compatibility as much as possible at the cost of extra lines in the XML.

**Comment by 3nids** [comment]
I still need to test if this correctly works (saving/reading projects and across versions), but this is ready for review API wise.

File: src/core/symbology/qgsmarkersymbollayer.cpp [hunk 2]
```diff
@@ -2651,15 +2655,70 @@ QgsSymbolLayer *QgsRasterMarkerSymbolLayer::create( const QgsStringMap &props )
...
-  QgsStringMap::iterator it = properties.find( QStringLiteral( "name" ) );
+  QVariantMap::iterator it = properties.find( QStringLiteral( "name" ) );
   if ( it != properties.end() )
   {
-    if ( saving )
-      it.value() = QgsSymbolLayerUtils::svgSymbolPathToName( it.value(), pathResolver );
-    else
-      it.value() = QgsSymbolLayerUtils::svgSymbolNameToPath( it.value(), pathResolver );
+    QVariant &value = it.value();
+    switch ( value.type() )
```
Comment (nyalldawson) [inline src/core/symbology/qgsmarkersymbollayer.cpp:2664]: Can you explain the situation here?

File: src/core/symbology/qgssymbollayerutils.cpp (hunk 1 above)
Comment (nyalldawson) [reply src/core/symbology/qgssymbollayerutils.cpp:2980]: That's the right move, thanks

**Comment by nyalldawson** [comment]
Looking great! This is definitely the right long-term move...

File: src/core/symbology/qgsmarkersymbollayer.cpp (hunk 2 above)
Comment (3nids) [reply src/core/symbology/qgsmarkersymbollayer.cpp:2664]: fixed.
I badly looked at the original code and thought this would require recursion.

**Pull request merged**